File: desktop/electron/notifications.ts
import { Notification } from 'electron';
import type { ContentType } from './ipc-types';

export class Notifier {
  private enabled: boolean;
  constructor(enabled: boolean) {
    this.enabled = enabled;
  }
  setEnabled(v: boolean): void {
    this.enabled = v;
  }
  notifyCapture(ct: ContentType, preview: string): void {
    if (!this.enabled) return;
    const titleMap: Record<ContentType, string> = {
      text: 'Text captured',
      link: 'Link captured',
      code: 'Code captured',
      color: 'Color captured',
      emoji: 'Emoji captured',
      image: 'Image captured',
      file: 'File path captured',
    };
    new Notification({
      title: titleMap[ct],
      body: preview.slice(0, 140),
      silent: true,
    }).show();
  }
}

File: desktop/src-tauri/src/notifications.rs
use crate::clipboard::ContentType;
use notify_rust::Notification;
use std::sync::atomic::{AtomicBool, Ordering};
use std::sync::Arc;

pub struct Notifier {
    enabled: Arc<AtomicBool>,
}

impl Notifier {
    pub fn new(enabled: bool) -> Self {
        Self {
            enabled: Arc::new(AtomicBool::new(enabled)),
        }
    }
    pub fn set_enabled(&self, v: bool) {
        self.enabled.store(v, Ordering::Relaxed);
    }

    pub fn notify_capture(&self, ct: ContentType, preview: &str) {
        if !self.enabled.load(Ordering::Relaxed) {
            return;
        }
        let summary = match ct {
            ContentType::Text => "Text captured",
            ContentType::Link => "Link captured",
            ContentType::Code => "Code captured",
            ContentType::Color => "Color captured",
            ContentType::Emoji => "Emoji captured",
            ContentType::Image => "Image captured",
            ContentType::File => "File path captured",
        };
        let trimmed: String = preview.chars().take(140).collect();
        let _ = Notification::new()
            .summary(summary)
            .body(&trimmed)
            .appname("Clippy")
            .timeout(2500)
            .show();
    }
}
